lib/constants/editor-interface-defaults/controls-defaults.ts

```diff
@@ -0,0 +1,194 @@
+  Symbol: { widgetId: 'singleLine' },
+  Integer: { widgetId: 'numberEditor' },
+  Number: { widgetId: 'numberEditor' },
+  Boolean: { widgetId: 'boolean' },
+  Date: { widgetId: 'datePicker' },
+  Location: { widgetId: 'locationEditor' },
+  Object: { widgetId: 'objectEditor' },
+  RichText: { widgetId: 'richTextEditor' },
+  Entry: { widgetId: 'entryLinkEditor' },
+  Asset: { widgetId: 'assetLinkEditor' },
+  Symbols: { widgetId: 'tagEditor' },
+  Entries: { widgetId: 'entryLinksEditor' },
+  Assets: { widgetId: 'assetLinksEditor' },
+  File: { widgetId: 'fileEditor' },
+}
+
+export const DEFAULTS_SETTINGS = {
+  Boolean: {
+    falseLabel: 'No',
+    helpText: null,
+    trueLabel: 'Yes',
+  },
+  Date: {
+    helpText: null,
+    ampm: '24',
+    format: 'timeZ',
+  },
+  Entry: {
+    helpText: null,
+    showCreateEntityAction: true,
+    showLinkEntityAction: true,
+  },
+  Asset: {
+    helpText: null,
+    showCreateEntityAction: true,
+    showLinkEntityAction: true,
+  },
+
+  Entries: {
+    helpText: null,
+    bulkEditing: false,
+    showCreateEntityAction: true,
+    showLinkEntityAction: true,
+  },
+  Assets: {
+    helpText: null,
+    showCreateEntityAction: true,
+    showLinkEntityAction: true,
+  },
+} as const
+
+interface DefaultWidget {
+  widgetId: string
+  settings?: {
+    helpText: null | string
+  }
+  fieldId: string
+  widgetNamespace: 'builtin'
+}
+
+function getDefaultWidget(field: keyof typeof DEFAULTS_WIDGET, fieldId: string) {
+  const defaultWidget: DefaultWidget = {
+    ...(DEFAULTS_WIDGET[field] as Pick<DefaultWidget, 'widgetId'>),
+    settings: {
+      helpText: null,
+    },
+    widgetNamespace: 'builtin',
+    fieldId,
+  }
+  if (in_(field, DEFAULTS_SETTINGS)) {
+    defaultWidget.settings = {
+      ...defaultWidget.settings,
+      ...DEFAULTS_SETTINGS[field],
+    }
+  }
+  return defaultWidget
+}
+
+// Given our internal identifier returns a minimal API field object.
+export function toApiFieldType(internal: keyof typeof INTERNAL_TO_API) {
+  return INTERNAL_TO_API[internal]
+}
+
+/*
+ * Gets the default widget ID for a field:
+ * - If a field allows predefined values then `dropdown` widget is used
+ *   in the presence of the `in` validation.
+ * - If a Text field is a title then the `singleLine` widget is used.
+ * - Otherwise a simple type-to-editor mapping is used.
+ */
+export default function getDefaultControlOfField(field: ContentFields, displayFieldId: string) {
+  let fieldType = toInternalFieldType(field)
+
+  if (!fieldType) return
+
+  const hasInValidation = (field.validations || []).find(
+    (v: ContentTypeFieldValidation) => 'in' in v
+  )
+
+  if (hasInValidation && DROPDOWN_TYPES.includes(fieldType)) {
+    return {
+      widgetId: 'dropdown',
+      fieldId: field.id,
+      widgetNameSpace: 'builtin',
+    }
+  }
+
+  const isTextField = fieldType === 'Text'
+  const isDisplayField = field.id === displayFieldId
+
+  if (isTextField && isDisplayField) {
```

Comment:
nit: does it make sense to combine these two together? Looks like we're using both only once so wont hurt to say `isDisplayTextField`